### DISCUSSION TITLE
Google Sheets connection in Tutorial

### DISCUSSION THREAD

**Discussion by indavidjool** [lowdefy/lowdefy#1099] [Docs]
The process for generating the service account key on the Google Console must have changed since tutorial was written.  I have figured out how to generate the key, but I get: Request Error: Google API error - [403] The caller does not have permission.  As far as I can tell it appears that the problem is that in order to give service accounts access to the data, there is a need to  [delegate domain-wide authority ] (https://developers.google.com/identity/protocols/oauth2/service-account) or have I missed a critical step?

**Comment by pleabargain** [comment]
I can concur. The docs do not match the UI of https://console.cloud.google.com/

**Comment by JonBendtsen** [comment]
Did you ever find a working solution?

**Reply by JonBendtsen** [reply]
Maybe this is an answer? https://github.com/lowdefy/lowdefy/discussions/692

**Reply by JonBendtsen** [reply]
didn't work for me

**Reply by JonBendtsen** [reply]
I got it to work. I needed to create a service account

https://docs.lowdefy.com/GoogleSheet

"The GoogleSheet connection can be used to connect to a Google using an API key or a service account. An API key can only be used for read-only access to public sheets, so most applications should use a service account. A guide to creating a service account can be found in the [tutorial](https://docs.lowdefy.com/tutorial-requests)."

**Reply by JonBendtsen** [reply]
Here's the command I used to translate my json file to the base64 string i inserted in my .env

`echo -e $( jq .'private_key' file.json ) | tr -d '"' | base64`

when you `jq` the file, you see that it has some **\n** so I echo that to translate the string **\n** into "real" newlines. I do this because while searching for the error message I got previously
  ```Request Error: error:0909006C:PEM routines:get_name:no start line```
then i saw someone writing something about exactly **\n** so I decided to just remove them.

Then there are the **`"`** I removed those with `tr` because the lowdefy tutorial in Step 5.11 said in the paste box **`Paste your key here (without quotes)`**
https://docs.lowdefy.com/tutorial-requests

Finally the `base64` command will base64 encode it - and it works for me, I get access and my app has put some data from a form into the Google Sheet.
